modules/afprog/afprog.c

```diff
@@ -358,18 +430,55 @@ afprogram_dd_init(LogPipe *s)
     }
   log_pipe_append(&self->super.super.super, (LogPipe *) self->writer);
 
-  return afprogram_dd_reopen(self);
+  return restore_successful ? TRUE : afprogram_dd_reopen(self);
+}
+
+static inline void
+afprogram_dd_store_reload_info(AFProgramDestDriver * const self, GlobalConfig * const cfg)
+{
```

Comment:
please attach the asterisk to the variable and use const as the qualifier of the type.

e.g. const AFProgramDestDriver *self, and const GlobalConfig *cfg)